test/pycheck/utils.py

```diff
@@ -281,31 +304,48 @@ def dsql(self, query, **kwargs):
         """Run a DuckDB query using duckdb.query()"""
         return self.sql(f"SELECT * FROM duckdb.query($ddb$ {query} $ddb$)", **kwargs)
 
-    def wait_until_table_exists(self, table_name, timeout=5):
-        end = time.time() + timeout
-        # Don't log these expected errors to the postgres log
-        self.sql("SET log_min_messages TO FATAL")
-        while time.time() < end:
-            try:
-                self.sql("SELECT %s::regclass", (table_name,))
-                self.sql("RESET log_min_messages")
-                return
-            except psycopg.errors.UndefinedTable:
-                time.sleep(0.1)
-        raise TimeoutError(f"Table {table_name} did not appear in time")
+    def wait_until_table_exists(self, table_name, timeout=5, **kwargs):
+        self.wait_until(
+            lambda: self.try_sql(
+                "SELECT %s::regclass",
+                (table_name,),
+                psycopg.errors.UndefinedTable,
+                **kwargs,
+            ),
+            f"Table {table_name} did not appear in time",
+            timeout=timeout,
+        )
 
-    def wait_until_schema_exists(self, schema_name, timeout=5):
+    def wait_until_schema_exists(self, schema_name, timeout=5, **kwargs):
+        self.wait_until(
+            lambda: self.try_sql(
```

Comment:
nit: Instead of waiting for a predicate, how about we wait until no exception occurs. Then we can make this a context manager, instead of having to use lambdas. (in case you haven't noticed. I much prefer context managers over lambdas in python)